src/main/java/com/mraof/minestuck/blockentity/machine/AnthvilBlockEntity.java

```diff
@@ -0,0 +1,259 @@
+	}
+	
+	@Override
+	public void saveAdditional(CompoundTag compound)
+	{
+		super.saveAdditional(compound);
+		
+		this.progressTracker.save(compound);
+		
+		compound.putShort("fuel", fuel);
+	}
+	
+	@Override
+	protected ItemStackHandler createItemHandler()
+	{
+		return new MachineProcessBlockEntity.CustomHandler(2, (index, stack) -> index == 0 || stack.is(ExtraForgeTags.Items.URANIUM_CHUNKS));
+	}
+	
+	@Override
+	protected void tick()
+	{
+		this.progressTracker.tick(this::processContents);
+	}
+	
+	private boolean contentsValid()
+	{
+		if(level.hasNeighborSignal(this.getBlockPos()))
+		{
+			return false;
+		}
+		
+		ItemStack fuel = itemHandler.getStackInSlot(1);
+		ItemStack input = itemHandler.getStackInSlot(0);
+		
+		boolean goodOnFuel = canBeRefueled() && fuel.is(ExtraForgeTags.Items.URANIUM_CHUNKS);
+		return goodOnFuel || !input.isEmpty();
+	}
+	
+	/**
+	 * Only handles refreshing uranium
+	 */
+	private void processContents()
+	{
+		if(canBeRefueled())
+		{
+			//checks for a uranium itemstack in the right item slot, increases the fuel value if some is found and then removes one count from the fuel stack
+			if(itemHandler.getStackInSlot(1).is(ExtraForgeTags.Items.URANIUM_CHUNKS))
+			{
+				//Refill fuel
+				addFuel((short) FUEL_INCREASE);
+				itemHandler.extractItem(1, 1, false);
+			}
+		}
+	}
+	
+	public static void attemptMend(AnthvilBlockEntity anthvil, ServerPlayer player)
+	{
+		Level level = anthvil.level;
+		ItemStack slotStack = anthvil.itemHandler.getStackInSlot(0);
+		GristCache playerCache = GristCache.get(player);
+		
+		ImmutableGristSet pickedGrist = mendingGrist(anthvil, playerCache, level, slotStack);
+		
+		if(!GristAmount.EMPTY.equals(pickedGrist))
+		{
+			if(slotStack.hasCraftingRemainingItem())
+			{
+				anthvil.itemHandler.setStackInSlot(0, slotStack.getCraftingRemainingItem());
+			} else
+			{
+				//amount of repair changes with the value of the grist, 10 at base
+				int repairAmount = (int) (10 * pickedGrist.asAmounts().get(0).getValue());
+				slotStack.setDamageValue(slotStack.getDamageValue() - repairAmount);
+				
+				if(!player.isCreative() && playerCache.tryTake(pickedGrist, GristHelper.EnumSource.CLIENT))
+				{
+					anthvil.fuel -= MEND_FUEL_COST;
+				}
+			}
+		}
+	}
+	
+	/**
+	 * Checks that there is enough fuel energy/grist for the machine to work and that there is something valid to mend, returns the appropriate grist amount if so
+	 */
+	private static ImmutableGristSet mendingGrist(AnthvilBlockEntity anthvil, GristCache playerCache, Level level, ItemStack slotStack)
+	{
+		if(level != null && !slotStack.isEmpty() && slotStack.isRepairable() && slotStack.isDamageableItem() && slotStack.isDamaged())
+		{
+			GristSet fullSet = GristCostRecipe.findCostForItem(slotStack, null, false, level);
+			
+			if(fullSet != null && !fullSet.isEmpty())
+			{
+				GristAmount pickedGrist = getUsedGrist(fullSet);
+				
+				if(anthvil.fuel >= MEND_FUEL_COST && playerCache.canAfford(pickedGrist))
```

Comment:
Too many responsibilities in this function. I suggest separating the conditions for if the machine can repair the item from the code that looks up the repair cost, moving some of the conditions into `attemptMend()`.

Reply:
Migrated parts of code to attemptMend and incorporated them as an early return checks